Tests/WorkspaceTests/ToolsVersionSpecificationRewriterTests.swift

```diff
@@ -17,185 +17,170 @@
+            toolsVersion: version,
+            fileSystem: inMemoryFileSystem
+        )
 
-        // Try to write a version with prerelease and build meta data.
-        let toolsVersion = ToolsVersion(version: "4.1.2-alpha.beta+sha.1234")
-        rewriteToolsVersionSpecificationToDefaultManifest(
-            content: "let package = ... \n",
-            version: toolsVersion
-        ) { result in
-            XCTAssertEqual(result, "// swift-tools-version:4.1.2\nlet package = ... \n")
-        }
+        // resultHandler(try inMemoryFileSystem.readFileContents(manifestFilePath))
+        let actual = try #require(try inMemoryFileSystem.readFileContents(manifestFilePath))
+        #expect(actual.validDescription == expected, "Actual is not expected")
     }
-    
-    func testManifestAccessFailures() throws {
+
+    @Test
+    func manifestAccessFailures() throws {
         let toolsVersion = ToolsVersion.v5_3
-        
+
         let inMemoryFileSystem = InMemoryFileSystem()
         let manifestFilePath = AbsolutePath("/pkg/Package.swift/Package.swift")
         try inMemoryFileSystem.createDirectory(manifestFilePath.parentDirectory, recursive: true) // /pkg/Package.swift/
-        
+
         // Test `ManifestAccessError.Kind.isADirectory`
-        XCTAssertThrowsError(
+
+        #expect{
             try ToolsVersionSpecificationWriter.rewriteSpecification(
                 manifestDirectory: manifestFilePath.parentDirectory.parentDirectory, // /pkg/
                 toolsVersion: toolsVersion,
                 fileSystem: inMemoryFileSystem
-            ),
-            "'/pkg/Package.swift' is a directory, and an error should've been thrown"
-        ) { error in
-            guard let error = error as? ToolsVersionSpecificationWriter.ManifestAccessError else {
-                XCTFail("a ManifestAccessError should've been thrown")
-                return
-            }
-            XCTAssertEqual(
-                error.kind,
-                .isADirectory
             )
-            XCTAssertEqual(
-                error.description,
-                "no accessible Swift Package Manager manifest file found at '\(manifestFilePath.parentDirectory)'; the path is a directory; a file is expected" // /pkg/Package.swift/
+        } throws: { error in
+            let error = try #require(
+                error as? ToolsVersionSpecificationWriter.ManifestAccessError,
+                "a ManifestAccessError should've been thrown"
             )
+            let isExpectedKind = (error.kind == .isADirectory)
+            let isExpectedDescription = (error.description == "no accessible Swift Package Manager manifest file found at '\(manifestFilePath.parentDirectory)'; the path is a directory; a file is expected")
+
+            return isExpectedKind && isExpectedDescription
         }
-        
+
         // Test `ManifestAccessError.Kind.noSuchFileOrDirectory`
-        XCTAssertThrowsError(
+        #expect {
```

Comment:
As above

Reply:
Discussing in the other thread.

Reply:
follow the conversation [on this comment thread](https://github.com/swiftlang/swift-package-manager/pull/8092#discussion_r1836907186).